### PR TITLE
Fix price field currency symbol position

### LINKED ISSUE #35558: [Enhancement]: Fix price field currency symbol position
### Describe the solution you'd like

We're looking to simplify the price input field and make it more usable by updating the layout, structure, and some interactions.

https://user-images.githubusercontent.com/79307566/201509568-d689ca80-54bb-433e-847b-a891ab7f065c.mov

**Acceptance criteria**

- [x] Applies to all price input fields in the experience.
- [x] Displayed currency unit follows the store's locale.
- [x] Currency symbol is fixed on the left side of the price input field.
- [x] Currency symbol has a different color from the input field value (`Gutenberg-700`).
- [x] Input value is aligned right.
- [x] Field formatting respects the store's setting, including thousand and decimal separators.
- [x] On blur, commas are added when the user enters a value. For example, when the user types `1234567` and leaves the field, we display the value as `1,234,567`.
- [x] On focus, we highlight the field's content so the user can quickly clear or overwrite the placeholder text.
- [x] On focus, the user can press the up and down arrow keys to increase or decrease the value by 1 (only concerns the value before the first comma, e.g. `0,5 -> 1.5`, NOT `0,5 -> 0.6`).

### Describe alternatives you've considered

_No response_

### Additional context

Related to #35180

### PR DESCRIPTION
**Pull request by mdperez86**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/35558

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Go to `Products -> Add New (MVP)`
2. Any price input field should has
3. Currency unit follows the store's locale
4. The symbol aligned always left
5. The symbol color now is `gray-700`
6. The value is aligned right
7. Field formatting respects the store's setting, including thousand and decimal separators
8. On blur, commas are added when the user enters a value
9. On focus, we highlight the field's content
10. On focus, the user can press the up and down arrow keys to increase or decrease the value by 1

<video src="https://user-images.githubusercontent.com/13334210/203852582-01598c36-46a8-4a7c-b55b-9cdbca9b5329.mov"></video>

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by AnnaMag — DISMISSED** [review]
LGTM :shipit:

File: plugins/woocommerce-admin/client/products/sections/pricing-section.tsx [hunk 1]
```diff
@@ -148,11 +148,45 @@ export const PricingSection: React.FC = () => {
...
+		},
+		onKeyUp( event: React.KeyboardEvent< HTMLInputElement > ) {
+			const name = event.currentTarget.name as keyof Pick<
+				Product,
+				'regular_price' | 'sale_price'
+			>;
+			const amount = Number.parseFloat(
+				sanitizePrice( values[ name ] || '0' )
+			);
+			const step = Number( event.currentTarget.step || '1' );
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/sections/pricing-section.tsx:178]: Nice!

File: plugins/woocommerce-admin/client/products/sections/utils.ts [hunk 2]
```diff
@@ -81,3 +81,5 @@ export const formatCurrencyDisplayValue = (
 
 	return value === undefined ? value : format( value ).replace( regex, '' );
 };
+
+export function stepUp() {}
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/sections/utils.ts]: @mdperez86 I assume this can be deleted and was maybe added by accident?

**Review by louwie17 — CHANGES_REQUESTED** [review]
LGTM 🚀 Nice work! I did leave one comment about the addition of an empty function.

File: plugins/woocommerce-admin/client/products/sections/utils.ts (hunk 2 above)
Comment (mdperez86) [reply plugins/woocommerce-admin/client/products/sections/utils.ts]: Yes, you are right! We don't need this at all. Thanks!!!

Comment (mdperez86) [reply plugins/woocommerce-admin/client/products/sections/utils.ts]: Solved -> [0c36e0b](https://github.com/woocommerce/woocommerce/pull/35718/commits/0c36e0b19c154b9a67cad525cf4bfd7b2b96fd1e)

**Review by louwie17 — APPROVED** [review]
Thanks for the quick fix, looks good to me ✔️ 

Also I guess @AnnaMag had beat me to this PR, sorry for the double review.

**Comment by mdperez86** [comment]
> Thanks for the quick fix, looks good to me ✔️
> 
> Also I guess @AnnaMag had beat me to this PR, sorry for the double review.

More eyes can always see more!!! So happy to receive feedback from different sources and point of views.

**Pull request merged**